app/ui/src/app/integration/import-export/import/integration-import.effects.ts

```diff
@@ -0,0 +1,34 @@
+import { Injectable } from '@angular/core';
+import { Action } from '@ngrx/store';
+import { Actions, Effect } from '@ngrx/effects';
+import { Observable } from 'rxjs/Observable';
+
+import { IntegrationImportService } from './integration-import.service';
+import {
+  IntegrationImportActions,
+  IntegrationUpload,
+  IntegrationUploadComplete,
+  IntegrationImport,
+  IntegrationImportComplete
+} from './integration-import.actions';
+
+@Injectable()
+export class IntegrationImportEffects {
+  @Effect()
+  importIntegration$: Observable<Action> = this.actions$
+    .ofType<IntegrationUpload>(IntegrationImportActions.IMPORT)
```

Comment:
You can spare importing `IntegrationUpload ` and use `IntegrationImportActions .IntegrationUpload ` here instead. It might be more verbose but will allow you to not to keep track and import each and every action created in the future.